service/dynamodb/expression/condition.go

```diff
@@ -63,45 +207,131 @@ func (cond ConditionBuilder) BuildExpression() (Expression, error) {
 	return expr, nil
 }
 
-// buildCondition will iterate over the tree of ConditionBuilders and
-// OperandBuilders and build a tree of ExprNodes
+// buildCondition will build a tree structure of ExprNodes based on the tree
+// structure of the input ConditionBuilder's child Conditions/Operands.
 func (cond ConditionBuilder) buildCondition() (ExprNode, error) {
 	switch cond.Mode {
 	case EqualCond:
 		return compareBuildCondition(cond)
+	case AndCond:
+		return compoundBuildCondition(cond)
+	case BetweenCond:
+		return betweenBuildCondition(cond)
 	}
 	return ExprNode{}, fmt.Errorf("No matching Mode to %v", cond.Mode)
 }
 
 // compareBuildCondition is the function to make ExprNodes from Compare
-// ConditionBuilders
+// ConditionBuilders. There will first be checks to make sure that the input
+// ConditionBuilder has the correct format.
 func compareBuildCondition(c ConditionBuilder) (ExprNode, error) {
-	if len(c.conditionList) != 0 {
-		return ExprNode{}, fmt.Errorf("Invalid ConditionBuilder. Expected 0 ConditionBuilders")
+	childNodes, err := buildChildNodes(c, 2, 0)
+	if err != nil {
+		return ExprNode{}, err
+	}
+	ret := ExprNode{
+		children: childNodes,
 	}
 
-	if len(c.operandList) != 2 {
-		return ExprNode{}, fmt.Errorf("Invalid ConditionBuilder. Expected 2 Operands")
+	// Create a string with special characters that can be substituted later: $c
+	switch c.Mode {
+	case EqualCond:
+		ret.fmtExpr = "$c = $c"
 	}
 
-	operandExprNodes := make([]ExprNode, 0, len(c.operandList))
-	for _, ope := range c.operandList {
-		exprNodes, err := ope.BuildOperand()
-		if err != nil {
-			return ExprNode{}, err
+	return ret, nil
+}
+
+// compoundBuildCondition is the function to make ExprNodes from And/Or
+// ConditionBuilders. There will first be checks to make sure that the input
+// ConditionBuilder has the correct format.
+func compoundBuildCondition(c ConditionBuilder) (ExprNode, error) {
+	childNodes, err := buildChildNodes(c, 0, 2)
+	if err != nil {
+		return ExprNode{}, err
+	}
+	ret := ExprNode{
```

Comment:
Initializations of these `ExprNode` and getting children probably can be moved to a helper since the setup is always the same.